src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -619,4 +619,17 @@ public function getReturnStatementsAst()
 
         return $visitor->getReturnNodes();
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
+    {
+        $objectType = (new FindTypeFromAst())->__invoke(
+            $this->reflector->getContextFactory()->createForNamespace(
```

Comment:
> but the library must cater for those with multiple classes.

It would do - just that each reflection class would have it's own context factory and not benefit from a shared cache.

> What I was suggesting is instantiate the [Cached]ContextFactory directly in the type finder classes

This would mean that each TypeFinder would have it's own CachedContextFactory, this would not convey any performance benefit (?)